File: 1813 Sentence Similarity III/1813.cpp
class Solution {
public:
    deque<string> split(string s){
        deque<string> words;
        istringstream ss(s);
        string word;

        while (ss >> word){
            words.push_back(word);
        }

        return words;
    }

    bool areSentencesSimilar(string s1, string s2) {
        deque<string> w1 = split(s1);
        deque<string> w2 = split(s2);
        
        while (!w1.empty() && !w2.empty() && w1.front() == w2.front()){
            w1.pop_front();
            w2.pop_front();
        }

        while (!w1.empty() && !w2.empty() && w1.back() == w2.back()){
            w1.pop_back();
            w2.pop_back();
        }

        return w1.empty() || w2.empty();
    }
};

File: 1813 Sentence Similarity III/1813.py
class Solution:
    def areSentencesSimilar(self, s1: str, s2: str) -> bool:
        w1 = deque(s1.split())
        w2 = deque(s2.split())
        
        while w1 and w2 and w1[0] == w2[0]:
            w1.popleft()
            w2.popleft()

        while w1 and w2 and w1[-1] == w2[-1]:
            w1.pop()
            w2.pop()

        return not w1 or not w2
